arbnode/dataposter/storage_test.go

```diff
@@ -0,0 +1,256 @@
+				}
+			})
+		}
+	}
+}
+
+func TestGetLast(t *testing.T) {
+	for name, s := range storages(t) {
+		t.Run(name, func(t *testing.T) {
+			ctx := context.Background()
+			for i := 0; i < 100; i++ {
+				val := strconv.Itoa(i)
+				if err := s.Put(ctx, uint64(i), nil, &val); err != nil {
+					t.Fatalf("Error putting a key/value: %v", err)
+				}
+				got, err := s.GetLast(ctx)
+				if err != nil {
+					t.Fatalf("Error getting a last element: %v", err)
+				}
+				if *got != val {
+					t.Errorf("GetLast() = %q want %q", *got, val)
+				}
+			}
+		})
+	}
+}
+
+func TestPrune(t *testing.T) {
+	ctx := context.Background()
+	for _, tc := range []struct {
+		desc      string
+		pruneFrom uint64
+		want      []*string
+	}{
+		{
+			desc:      "prune all elements",
+			pruneFrom: 20,
+		},
+		{
+			desc:      "prune all but one",
+			pruneFrom: 19,
+			want:      strPtrs([]string{"19"}),
+		},
+		{
+			desc:      "pruning first element",
+			pruneFrom: 1,
+			want: strPtrs([]string{"1", "2", "3", "4", "5", "6", "7",
+				"8", "9", "10", "11", "12", "13", "14", "15", "16", "17", "18", "19"}),
+		},
+		{
+			desc:      "pruning first 11 elements",
+			pruneFrom: 11,
+			want:      strPtrs([]string{"11", "12", "13", "14", "15", "16", "17", "18", "19"}),
+		},
+		{
+			desc:      "pruning from higher than biggest index",
+			pruneFrom: 30,
+			want:      strPtrs([]string{}),
+		},
+	} {
+		// Storages must be re-initialized in each test-case.
+		for name, s := range initStorages(ctx, t) {
+			t.Run(name+"_"+tc.desc, func(t *testing.T) {
+				if err := s.Prune(ctx, tc.pruneFrom); err != nil {
+					t.Fatalf("Prune(%d) unexpected error: %v", tc.pruneFrom, err)
+				}
+				got, err := s.GetContents(ctx, 0, 20)
+				if err != nil {
+					t.Fatalf("GetContents() unexpected error: %v", err)
+				}
+				if diff := cmp.Diff(tc.want, got); diff != "" {
+					t.Errorf("Prune(%d) unexpected diff:\n%s", tc.pruneFrom, diff)
+				}
+			})
+		}
+	}
+}
+
+func TestLength(t *testing.T) {
+	ctx := context.Background()
+	for _, tc := range []struct {
+		desc      string
+		pruneFrom uint64
+	}{
+		{
+			desc: "not prune any elements",
+		},
+		{
+			desc:      "prune all but one",
+			pruneFrom: 19,
+		},
+		{
+			desc:      "pruning first element",
+			pruneFrom: 1,
+		},
+		{
+			desc:      "pruning first 11 elements",
+			pruneFrom: 11,
+		},
+		{
+			desc:      "pruning from higher than biggest index",
+			pruneFrom: 30,
+		},
+	} {
+		// Storages must be re-initialized in each test-case.
+		for name, s := range initStorages(ctx, t) {
+			t.Run(name+"_"+tc.desc, func(t *testing.T) {
+				if err := s.Prune(ctx, tc.pruneFrom); err != nil {
+					t.Fatalf("Prune(%d) unexpected error: %v", tc.pruneFrom, err)
+				}
+				got, err := s.Length(ctx)
+				if err != nil {
+					t.Fatalf("Length() unexpected error: %v", err)
+				}
+				if want := arbmath.MaxInt(0, 20-int(tc.pruneFrom)); got != want {
+					t.Errorf("Length() = %d want %d", got, want)
+				}
+			})
+		}
+
+	}
+}
```

Comment:
It doesn't look like any of these tests test overwriting elements. We should add a test that does that.

Reply:
Added test for it.